example/example.js

```diff
@@ -0,0 +1,63 @@
+'use strict';
+
+import config from 'config';
+import {Server} from 'hapi';
+import good from 'good';
+import goodConsole from 'good-console';
+import hapiRoutes from '../es6/hapi-mysql-routes-plugin';
+import Joi from 'joi';
+import pkg from './package.json';
+
+const port = 9000;
+const server = new Server();
+
+server.connection({ port: port, labels: ['api'] });
+
+const api = server.select('api');
+
+let listQueryValidationSchema = {
+  name: Joi.string().optional()
+};
+
+api.register(
+  [
+    {
+      register: good,
+      options: {
+        reporters: [{
+          reporter: goodConsole,
+          events: {
+            request: '*',
+            response: '*',
+            log: '*',
+            error: '*'
+          }
+        }],
+        requestPayload: true,
+        responsePayload: true
+      }
+    },
+    {
+      register: hapiRoutes,
+      options: {
+        version: pkg.version,
+        tags: ['api'],
+        mysqlConfig: config.mysql, //required
+        tableName: 'unique_object', //required
+        index: 'uid', //required
+        tableHeaderFormat: 'snakeCase', //required - another value - camelCase
+        validateListQuerySchema: listQueryValidationSchema //optional - this
```

Comment:
This is not a current requirement I just want to ask a question about using this module. It's something to be aware of on any module that defines routes.

Hapijs routes are highly configurable, and people like to do all sorts of things to them. When providing a plugin that houses special routes, it's nice to let the user pass in their own custom route configurations so they can do things like "auth", "validation", "cors", and all that other crazy customizable stuff hapi can do. The way you have it set up right now boxes in all the functionality for these hapi routes and doesn't allow people to extend things for themselves. Perhaps we should consider an approach where the user can send in their own special route configuration for each route which further extends the basic configuration the plugin sets up for them by default.